framework/src/utils/MooseEnumBase.C

```diff
@@ -127,3 +105,96 @@ MooseEnumBase::checkDeprecatedBase(const std::string & name_upper) const
       mooseWarning(name_upper + " is deprecated");
   }
 }
+
+void
+MooseEnumBase::addEnumerationName(const std::string & raw_name)
+{
+  // Make sure the option is not malformed
+  if (raw_name.find_first_of('=') == 0 || raw_name.find_last_of('=') == raw_name.length() - 1)
+    mooseError("You cannot place whitespace around the '=' character in MooseEnumBase");
+
+  // Split on equals sign
+  std::vector<std::string> name_value;
+  MooseUtils::tokenize(MooseUtils::trim(raw_name), name_value, 1, "=");
+
+  // There should be one or two items in the name_value
+  if (name_value.size() < 1 || name_value.size() > 2)
+    mooseError("Invalid option supplied in MooseEnumBase: ", raw_name);
+
+  // Remove un-wanted space around string
+  name_value[0] = MooseUtils::trim(name_value[0]);
+
+  // See if there is a value supplied for this option
+  // strtol allows for proper conversions of both int and hex strings
+  int value;
+  if (name_value.size() == 2)
+    value = strtol(name_value[1].c_str(), NULL, 0);
+  else if (_ids.empty())
+    value = 0;
+  else
+    value = *_ids.rbegin() + 1;
+
+  addEnumerationName(name_value[0], value);
+}
+
+void
+MooseEnumBase::addEnumerationName(const std::string & name, const int & value)
+{
+  // preserve case for raw options, append to list
+  if (!_raw_names.empty())
+    _raw_names += " ";
+  _raw_names += name;
+
+  // convert name to uppercase
+  std::string upper(name);
+  std::transform(upper.begin(), upper.end(), upper.begin(), ::toupper);
+
+  // check ids and names
+  if (_ids.find(value) != _ids.end())
+    mooseError("The id ", value, " already exists in the enumeration.");
+  if (_name_to_id.find(upper) != _name_to_id.end())
+    mooseError("The name ", upper, " already exists in the enumeration.");
+
+  // populate internal datastructures
+  _names.push_back(upper);
+  _ids.insert(value);
+  _name_to_id[upper] = value;
+  _id_to_name[value] = upper;
+  _name_to_raw_name[upper] = name + " "; // add space to make the name unique for removal
+}
+
+int
+MooseEnumBase::id(std::string name) const
```

Comment:
I think this could go away, I believe it is simply for printing out text rather than numbers in messages.